### DISCUSSION TITLE
Option for passing bookmark URL in as query parameter

### DISCUSSION THREAD

**Discussion by timshannon** [go-shiori/shiori#333] [Feature requests]
Since bookmarklet support has been dropped favor of the extension (#180), it be nice to have a fall back for any other scenario where the extension doesn't work (other browsers, locked down browser environments, etc).  

Simply accepting a URL parm at the root page that automatically opens the Add bookmark dialog and pre-populates the URL from the parameter value would allow for this.

`https://my-shiori-url.com?add_url=https%3A%2F%2Fgoogle.com`

I may add a PR for this, if you think it's a reasonable addition.  It should be able to be done entirely in the frontend Vue code.

**Comment by deanishe** [comment]
Bookmarklet support is important to me. It's simpler, and much more widely supported, than browser extensions.

**Comment by clach04** [comment]
I don't have a strong opinion on what the URL should be but #84 with commit 4b939379e83b34f6d9d97517a80b8dd44313bbe8 introduced `/submit?url=` - see:

* https://github.com/go-shiori/shiori/commit/4b939379e83b34f6d9d97517a80b8dd44313bbe8#diff-52f15f47c42ebad7b01d0c2eb2a495faR43
* https://github.com/go-shiori/shiori/commit/4b939379e83b34f6d9d97517a80b8dd44313bbe8#diff-68382b9aa5b82414a20bf208bb1b0859R654
* https://github.com/go-shiori/shiori/commit/4b939379e83b34f6d9d97517a80b8dd44313bbe8#diff-b15905fa00a4be61475ec51fae6c09f1R109

This was a fallback for non-https installations (I'm personally running a private instance withOUT https).

I'd recommend adopting the previous URL (and bookmarklet) support.

**Comment by deanishe** [comment]
> I'd recommend adopting the previous URL (and bookmarklet) support.

Was that removed? That's definitely going back in. Looks like it worked perfectly.

A GET-based submission endpoint is essential, imo. It allows you to integrate Shiori with almost anything.

**Comment by amitjethani** [comment]
I'd like to bump this request as well because it is such a huge QoL feature for Shiori.

**Comment by piegamesde** [comment]
I am not comfortable with the extension being able to read all the sites I visit. A bookmarklet would alleviate most of these concerns.

Furthermore, I'd like to have token based authentication instead of username and password, for increased security. This could work like following:

- An account has a token that is displayed somewhere in the UI
- There is a button to invalidate the token and generating a new one
- The token only permits to add new bookmarks and nothing else. With this, the worst thing that can happen to a lost token is spam.

**Reply by fmartingr** [reply]
hey @piegamesde ! Just as a comment, this is what I'm trying to aim for! Better API (which means better integrations), better authentication/sessions and tokens. But I'm just one person and I'm very busy lately. We will get there with time :)

**Comment by sommerper** [comment]
Is there an update on this? A bookmarklet feature would let me integrate this with iOS as well.

**Reply by fmartingr** [reply]
The issue tracking this is here: #269

I've put it in the [roadmap](https://github.com/orgs/go-shiori/projects/2/views/4?visibleFields=%5B%22Repository%22,%22Title%22,%22Assignees%22,%22Status%22,%22Labels%22,%22Linked+pull+requests%22,%22Milestone%22%5D) for 1.9.0, but we may end up doing it sooner.

I need to finish setting up the proper auth methods in order to work with the bookmarklet; for iOS, I'm assuming the API refactor would work as well, right? (you mean iOS Shortcuts or something else?)

**Reply by sommerper** [reply]
Sorry for the late reply. Yes, I mean Shortcuts (and bookmarklets). So I can just press share on a website and send the url to Shiori.
